pennylane/transforms/metric_tensor.py

```diff
@@ -425,14 +453,24 @@ def wrapper(*args, **kwargs):
             tkwargs["approx"] = "block-diag"
             return self(qnode, *targs, **tkwargs)(*args, **kwargs)
 
-        if old_interface == "auto":
-            qnode.interface = "auto"
-
         if not hybrid:
             return mt
 
         kwargs.pop("shots", False)
-        cjac = cjac_fn(*args, **kwargs)
+        # Special case where we apply a Jax transform (jacobian e.g.) on the gradient transform and argnums are
+        # defined on the outer transform and therefore on the args.
+        if argnums is None and argnums_ is None and interface == "jax":
+            cjac = qml.transforms.classical_jacobian(
+                qnode, argnum=qml.math.get_trainable_indices(args), expand_fn=self.expand_fn
+            )(*args, **kwargs)
+        else:
+            if interface == "jax":
+                argnum_cjac = argnums if argnums is not None else [0]
+            else:
+                argnum_cjac = None
+            cjac = qml.transforms.classical_jacobian(
+                qnode, argnum=argnum_cjac, expand_fn=self.expand_fn
+            )(*args, **kwargs)

```

Comment:
I feel like this is rather convoluted, so I'm wondering if the following is correct?
```suggestion
        if interface == "jax":
            argnum_cjac = trainable_params or argnums_
        else:
            argnum_cjac = argnums
        cjac = qml.transforms.classical_jacobian(
            qnode, argnum=argnum_cjac, expand_fn=self.expand_fn
        )(*args, **kwargs)
```
if jax, but trainable_params exist, then we don't go to the above block because that's what we want, right? If they don't exist, we go into the above code block, and ensure that `argnums_` has what we want. If not jax, we just use the previously-existing behaviour, which is using `argnums`, right?

This should apply to the other PR as well

Reply:
Thanks great simplification I just had to change your code a bit